trio/_unix_pipes.py

```diff
@@ -33,9 +33,11 @@ def __init__(self, fd: int):
         if not isinstance(fd, int):
             raise TypeError("file descriptor must be an int")
         self.fd = fd
-        # Flip the fd to non-blocking mode
-        flags = fcntl.fcntl(self.fd, fcntl.F_GETFL)
-        fcntl.fcntl(self.fd, fcntl.F_SETFL, flags | os.O_NONBLOCK)
+        # Store original state, and ensure non-blocking mode is enabled
+        self._original_flags = fcntl.fcntl(self.fd, fcntl.F_GETFL)
+        fcntl.fcntl(
+            self.fd, fcntl.F_SETFL, self._original_flags | os.O_NONBLOCK
+        )
```

Comment:
Let's use `os.get_blocking` and `os.set_blocking`. Advantages:

- Less fcntl line noise
- In the unlikely event that someone else is fiddling with other unrelated fcntl flags while the `FdStream` is open, we won't inadverdantly overwrite their changes when we reset the blocking state